### ISSUE TITLE
input forward cannot match tags in output

### ISSUE THREAD

**Issue by markuman** [fluent/fluent-bit#2680]
## Bug Report

**Describe the bug**

The local docker daemon is logging to fluent-bit using foward/fluentd.

```
"LogDriver": {
                    "Name": "fluentd",
                    "Options": {
                        "fluentd-address": "127.0.0.1:24224",
                        "fluentd-async-connect": "true",
                        "tag": "matomo.log"
                    }
                },

```

fluent-bit config

```
[INPUT]
    Name              forward
    Listen            127.0.0.1
    Port              24224
    Buffer_Chunk_Size 32KB
    Buffer_Max_Size   64KB

[FILTER]
    Name record_modifier
    Match *
    Record hostname ${HOSTNAME}

[FILTER]
    Name aws
    Match *
    az true
    ec2_instance_id true
    private_ip true
    account_id true
    vpc_id true

[OUTPUT]
    Name stdout
    Match matomo.log

[OUTPUT]
    Name cloudwatch_logs
    Match  matomo.log
    region eu-central-1
    log_group_name fluentd/matomo/access.log
    log_stream_prefix ${HOSTNAME}
    auto_create_group On

```

But it results in a no matching route for tag `matomo.log`

```
[2020/10/13 09:29:48] [ warn] [input chunk] no matching route for input chunk 1877-1602574188.152091056.flb
```

When I catch all in stdout  with a  `Match *`, it displays the correct tag.

```
[17] matomo.log: [1602574272.000000000, {"container_id"=>"7d06320d7006ad7f450cb63330e7dce321384d661feb49fb83110d27bfb779bb", "container_name"=>"/matomo.1.0ks1leyhuszx5ijpknqhr0n3g", "source"=>"stdout", "log"=>"10.115.2.223 - - [13/Oct/2020:07:31:12 +0000] "GET /plugins/Morpheus/images/logo.svg?matomo HTTP/1.1" 304 110 "https://matomo.....de/" "Mozilla/5.0 (X11; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0"", "hostname"=>"matomo", "az"=>"eu-central-1a", "ec2_instance_id"=>"i-0fc2e2ea0ebeb69f0", "private_ip"=>"10.115.3.165", "vpc_id"=>"vpc-d12e677e", "account_id"=>"1234567890"}]

```

**To Reproduce**
full fluent-bit config 

```
[SERVICE]
    Flush     1
    Daemon    off
    Log_Level debug
    storage.path              /var/log/flb-storage/
    storage.sync              normal
    storage.checksum          off
    storage.backlog.mem_limit 5M

[INPUT]
    Name        tail
    Path        /var/log/auth.log
    Tag         auth.log

[INPUT]
    Name        tail
    Path        /var/log/kern.log
    Tag         kern.log

[INPUT]
    Name        tail
    Path        /var/log/daemon.log
    Tag         daemon.log

[INPUT]
    Name        tail
    Path        /var/log/syslog
    Tag         sys.log

[INPUT]
    Name        tail
    Path        /var/log/messages
    Tag         messages.log

[INPUT]
    Name mem
    Interval_Sec  60
    Tag  mem

[INPUT]
    Name              forward
    Listen            127.0.0.1
    Port              24224
    Buffer_Chunk_Size 32KB
    Buffer_Max_Size   64KB

[OUTPUT]
    Name stdout
    Match matomo.log

[OUTPUT]
    Name cloudwatch_logs
    Match  matomo.log
    region eu-central-1
    log_group_name fluentd/matomo/access.log
    log_stream_prefix ${HOSTNAME}
    auto_create_group On

[FILTER]
    Name record_modifier
    Match *
    Record hostname ${HOSTNAME}

[FILTER]
    Name aws
    Match *
    az true
    ec2_instance_id true
    private_ip true
    account_id true
    vpc_id true

[OUTPUT]
    Name cloudwatch_logs
    Match  auth.log
    region eu-central-1
    log_group_name fluentd/auth.log
    log_stream_prefix ${HOSTNAME}
    auto_create_group On

[OUTPUT]
    Name cloudwatch_logs
    Match  sys.log
    region eu-central-1
    log_group_name fluentd/sys.log
    log_stream_prefix ${HOSTNAME}
    auto_create_group On

[OUTPUT]
    Name cloudwatch_logs
    Match  messages.log
    region eu-central-1
    log_group_name fluentd/messages.log
    log_stream_prefix ${HOSTNAME}
    auto_create_group On

[OUTPUT]
    Name cloudwatch_logs
    Match  kern.log
    region eu-central-1
    log_group_name fluentd/kern.log
    log_stream_prefix ${HOSTNAME}
    auto_create_group On

[OUTPUT]
    Name cloudwatch_logs
    Match  daemon.log
    region eu-central-1
    log_group_name fluentd/daemon.log
    log_stream_prefix ${HOSTNAME}
    auto_create_group On

[OUTPUT]
    Name cloudwatch_logs
    Match mem
    log_stream_name mem_${HOSTNAME}
    log_group_name fluent-bit-cloudwatch
    region eu-central-1
    log_format json/emf
    metric_namespace fluent-bit-metrics
    metric_dimensions ${HOSTNAME}
    auto_create_group true

```
**Expected behavior**

Tag `matomo.log` is Matched and also directed to cloudwatch logs

**Your Environment**

* Version used: 1.6.0  
* Configuration: see above
* Environment name and version (e.g. Kubernetes? What version?): Docker version 19.03.12, build 48a66213fe  
* Server type and version: EC2
* Operating System and version: Debian Buster  
* Filters and plugins: see above

**Comment by otakumike** [comment]
I believe I might have been experiencing the same issue where an exact Match tag didn't work, but adding a wildcard to the end of it would work. 

In my setup, I have the following configuration on the input server:
```
[SERVICE]
    Flush               1
    Daemon              off
    Log_Level           debug
    Parsers_File        /etc/td-agent-bit/parsers-custom.conf

[INPUT]
    Name                tail
    Path                /app/logs/cache.access.log
    DB                  /app/logs/cache.access.log.db
    Exclude_Path        *.gz
    Skip_Long_Lines     On
    Mem_Buf_Limit       64m
    Buffer_Chunk_Size   4k
    Buffer_Max_Size     1m
    Tag                 cache_access
    Parser              generic
```

Then on the receiving server, I had the following configuration: 
```
[SERVICE]
    Flush               1
    Daemon              off
    Log_Level           debug
    Parsers_File        /etc/td-agent-bit/parsers-custom.conf

[INPUT]
    Name                forward
    Listen              0.0.0.0
    Port                24224
    Mem_Buf_Limit       128m
    Buffer_Chunk_Size   256k
    Buffer_Max_Size     256k

[OUTPUT]
    Name                file
    Match               cache_access
    Path                /app/logs
    File                cache.access.log
    Format              template
    Template            {timestamp} {message}
```

This would result in the same error that markuman was seeing
`[ warn] [input chunk] no matching route for input chunk 88-1602799132.107260438.flb`

To test this, I set up a Match with `*`:
```
[OUTPUT]
    Name                file
    Match               *
    Path                /app/logs
    File                star.log
```

And sure enough I could see my `cache_access` tag being recorded to this file:
```
cache_access: [1602798715.705112545, {"timestamp":"2020-10-15T21:51:55+00:00","message":"...."}]
```

Finally, if I just added an asterisk to the end of my `cache_access` tag everything would work and the message would be routed correctly:
```
[OUTPUT]
    Name                file
    Match               cache_access*
    Path                /app/logs
    File                cache.access.log
    Format              template
    Template            {timestamp} {message}
```

I've checked and double-checked that there's no special characters or spaces around `cache_access`, and I can't see anything there that might cause this.

**Comment by edsiper** [comment]
Fixed in 8926785927f6431948c72ddf770d2dd925aba322
